File: L002_Basics/is_Prime.java
import java.util.Scanner;

public static class is_Prime {
    public static Scanner scn = new Scanner(System.in);

    public static boolean isPrime_(int n) {
        boolean prime = true;
        for (int i = 2; i * i <= n; i++) {
            if (n % i == 0) {
                prime = false;
                break;
            }
        }
        return prime;
    }

    public static void isPrime(int n) {

        boolean ans = isPrime_(n);
        if (ans)
            System.out.println("prime");
        else
            System.out.println("not prime");
    }

    public static void main(String[] args) {

        int t = scn.nextInt();

        while (t-- > 0) {
            int n = scn.nextInt();
            isPrime(n);
        }
    }
}

File: L002_Basics/is_Prime.cpp
#include <iostream>
using namespace std;

bool isPrime_(int n){
    bool prime = true;
    for(int i=2;i*i<=n;i++){
        if(n%i==0){
            prime = false;
            break;
        }   
    }
    return prime;
}

void isPrime(int n){
  
    if(n == 0 || n == 1) {
        cout<<"not prime"<<endl;
        return;
    }
    
    bool ans = isPrime_(n);
    if(ans) cout<<"prime"<<endl;
    else cout<<"not prime"<<endl;
}

int main(){
    int t;
    cin >> t;

    while ( t-- > 0){
        int n;
        cin>>n;
        isPrime(n);
    }
   
    
    return 0;
}